index.js

```diff
@@ -101,7 +102,16 @@ function requestAsEventEmitter(opts) {
 }
 
 function asPromise(opts) {
-	return new PCancelable((onCancel, resolve, reject) => {
+	const timeout = opts.gotTimeout && typeof opts.gotTimeout.request === 'number' ?
+		opts.gotTimeout.request :
+		typeof opts.gotTimeout === 'number' ?
+			opts.gotTimeout :
+			false;
+
+	return (requestPromise => timeout ?
+		pTimeout(requestPromise, timeout, new got.RequestError({message: 'Request timed out', code: 'ETIMEDOUT'}, opts)) :
+		requestPromise
```

Comment:
This could be more readable. Can you move it (line 111 to 113) out into a separate variable?